fast64_internal/oot/oot_level_writer.py

```diff
@@ -460,34 +460,66 @@ def ootProcessEmpties(scene, room, sceneObj, obj, transformMatrix):
 			actorProp = obj.ootActorProperty
-			addActor(room, OOTActor(getCustomProperty(actorProp, 'actorID'), 
-				translation, rotation, actorProp.actorParam,
-				None if not actorProp.rotOverride else (actorProp.rotOverrideX, actorProp.rotOverrideY, actorProp.rotOverrideZ)),
+			detailedProp = obj.fast64.oot.actor
+			actorID = actorProp.actorID
+			rotX = rotY = rotZ = ''
+			panelParams = getattr(detailedProp, "actorParam", "")
+
+			if panelParams != "":
+				# Figure out which rotation to export, Blender's or the override
+				if detailedProp.XRotBool or (actorID == 'Custom'):
+					rotX = getActorProperty(detailedProp, actorID, 'XRot')
+					if rotX is None: rotX = f'{rotation[0]}'
+				else: rotX = f'{rotation[0]}'
+
+				if detailedProp.YRotBool or (actorID == 'Custom'):
+					rotY = getActorProperty(detailedProp, actorID, 'YRot')
+					if rotY is None: rotY = f'{rotation[1]}'
+				else: rotY = f'{rotation[1]}'
+
+				if detailedProp.ZRotBool or (actorID == 'Custom'):
+					rotZ = getActorProperty(detailedProp, actorID, 'ZRot')
+					if rotZ is None: rotZ = f'{rotation[2]}'
+				else: rotZ = f'{rotation[2]}'
+
+			if panelParams == "" or (actorID == 'Custom' and detailedProp.rotOverrideCustom == False):
+				rotX = f'{rotation[0]}'
+				rotY = f'{rotation[1]}'
+				rotZ = f'{rotation[2]}'
+
+			addActor(room, OOTActor(getActorProperty(detailedProp, actorID, 'actorID'),
+				translation, rotation, getActorProperty(detailedProp, actorID, 'actorParam'), (rotX, rotY, rotZ)),
 				actorProp, "actorList", obj.name)
 		elif obj.ootEmptyType == "Transition Actor":
+			if rotation[0] != 0 or rotation[2] != 0:
+				raise PluginError("Transition Actors can't have X and Y Rotations!")
+
 			transActorProp = obj.ootTransitionActorProperty
+			detailedProp = obj.fast64.oot.actor
+			actorID = detailedProp.transActorID
+			if detailedProp.YRotBool or (actorID == 'Custom'):
+				rotY = getActorProperty(detailedProp, actorID, 'YRot')
+				if rotY is None: rotY = f'{rotation[1]}'
+			else: rotY = f'{int(round(rotation[1]))}'
+
 			addActor(scene, OOTTransitionActor(
-				getCustomProperty(transActorProp.actor, "actorID"),
+				getActorProperty(detailedProp, actorID, 'transActorID'),
 				room.roomIndex, transActorProp.roomIndex,
 				getCustomProperty(transActorProp, "cameraTransitionFront"),
 				getCustomProperty(transActorProp, "cameraTransitionBack"),
-				translation, rotation[1], # TODO: Correct axis?
-				transActorProp.actor.actorParam), transActorProp.actor, "transitionActorList", obj.name)
-			#scene.transitionActorList.append(OOTTransitionActor(
-			#	getCustomProperty(transActorProp.actor, "actorID"),
-			#	room.roomIndex, transActorProp.roomIndex,
-			#	getCustomProperty(transActorProp, "cameraTransitionFront"),
-			#	getCustomProperty(transActorProp, "cameraTransitionBack"),
-			#	translation, rotation[1], # TODO: Correct axis?
-			#	transActorProp.actor.actorParam))
+				translation, rotY, # TODO: Correct axis?
```

Comment:
```suggestion
				translation, rotY,
```
I assume this was about "is +y the upwards axis" and yes it is